app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\User;
+
+class UserController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        //
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show(Request $request, $id)
+    {
+
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        try {
+            $user = User::find($id);
+
+            return view('user.user-info', ['user' => $user]);            
+        } catch (Exception $e) {
+            return redirect('/');   
+        }
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $data = $request->only([
+            'fullname',
+            'nickname',
+            'image',
+        ]);
+
+        if (isset($data['image'])) {
+            $uploaded = $this->upload($data['image']);
+
+            if (!$uploaded['status']) {
+                return back()->with('status', $uploaded['msg']);
+            }
+
+            $data['image'] = $uploaded['file_name'];
+        }
+
+        try {
+            $user = User::find($id);
+            $user->update($data);   
+        } catch (Exception $e) {
```

Comment:
```suggestion
        } catch (\Exception $e) {
            \Log::error($e);
```
để như thế này hoặc là phải `use Exception` ở đầu file nhé